sqlglot/parser.py

```diff
@@ -1350,8 +1350,24 @@ def _parse_with(self, skip_with_token: bool = False) -> t.Optional[exp.Expressio
 
     def _parse_cte(self) -> exp.Expression:
         alias = self._parse_table_alias()
-        if not alias or not alias.this:
-            self.raise_error("Expected CTE to have alias")
+
+        # Alternative Clickhouse syntax: https://clickhouse.com/docs/en/sql-reference/statements/select/with/
+        if (
+            not alias
+            or not alias.this
+            or not self._match_pair(TokenType.ALIAS, TokenType.L_PAREN, advance=False)
```

Comment:
The `_match_pair` check here is so that we can parse `WITH (SELECT 1) AS foo`.

Reply:
not sure i get it

Reply:
Sorry, my bad. I got confused with the examples. What I meant to say is that it solves the following form:

```sql
WITH sum(bytes) AS foo SELECT foo FROM system.parts
```

Without this pair match check, `sum(bytes)` will be parsed as a table alias right above this branch and the parsing would fail later.